rust/src/delta_datafusion.rs

```diff
@@ -367,27 +383,86 @@ impl TableProvider for DeltaTable {
                 .cloned()
                 .collect(),
         ));
-        ParquetFormat::default()
+        let parquet_scan = ParquetFormat::default()
             .create_physical_plan(
                 FileScanConfig {
-                    object_store_url,
+                    object_store_url: self.storage.object_store_url(),
                     file_schema,
                     file_groups: file_groups.into_values().collect(),
                     statistics: self.datafusion_table_statistics(),
                     projection: projection.clone(),
                     limit,
                     table_partition_cols,
+                    config_options: Arc::new(Default::default()),
                 },
                 filters,
             )
-            .await
+            .await?;
+        let delta_scan = DeltaScan {
+            url: self.table_uri(),
+            parquet_scan: parquet_scan,
+        };
+
+        Ok(Arc::new(delta_scan))
     }
 
     fn as_any(&self) -> &dyn Any {
         self
     }
 }
 
+/// A wrapper for parquet scans that installs the required ObjectStore
```

Comment:
Note: this will likely also need to perform [column mapping](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#column-mapping) later when we support reader protocol v2.

Reply:
should we add a TODO comment for this?

Reply:
> should we add a TODO comment for this?

Done!